the-collab-lab/tcl-73-smart-shopping-list
# Create Delete Item button

## Description

We created a button that allows users to delete a single item from their list. This required us to write the delete function in the firebase.js file and incorporate a handleDelete hook in the ListItem react component. We also incorporated the JavaScript confirm() method requiring users to confirm that they truly want to delete the item before it is deleted from the database.

## Related Issue

Closes #12 

## Acceptance Criteria


## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

## Updates

enhancement, accessibility

### Before

![Screenshot 2024-03-12 at 7 39 25 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/135680082/5f04b458-da9a-47bb-a465-8d741ed07590)

### After

![Screenshot 2024-03-12 at 7 39 01 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/135680082/45f813c6-177e-48f2-a4a6-2e7fd873127d)

## Testing Steps / QA Criteria

- `git pull`
- `git checkout pc-rs-feature-delete-item`
- `npm ci`
- `npm start`
- navigate to the `List` page
- Observe the each list item and notice each one has a `Delete` button attached to it
- Select an item to delete, click the delete button and view the confirmation message
- Click `OK` on the confirmation box and then an alert will appear noting the item was successfully deleted
- Click `Cancel` on the confirmation box and notice that the item remains on the list, no other alert pops up
